resources/testharness.js

```diff
@@ -680,6 +681,23 @@ policies and contribution forms [3].
         });
     }
 
+    /**
+     * Similar to `promise_test`, but executing in parallel rather than
+     * sequentially.
+     *
+     * See RFC 75:
+     * https://github.com/web-platform-tests/rfcs/blob/master/rfcs/async_promise_test.md
+     */
+    function async_promise_test(func, name, properties) {
```

Comment:
This doesn't set `_is_promise_test` to true, so its cleanup behavior will be different (see https://web-platform-tests.org/writing-tests/testharness-api.html?highlight=promise_setup#cleanup, second paragraph). I'm not really sure why cleanup functions have this extra behavior for promise tests - @jgraham ? (I.e. maybe we should just let all cleanup functions return promises?)

Reply:
https://bocoup.com/blog/the-eight-month-omelette-adding-a-feature-to-one-million-conformance-tests is some relevatn history here.